inc/functions.php

```diff
@@ -117,3 +117,146 @@ function wp_sitemaps_get_max_urls( $object_type ) {
 	 */
 	return apply_filters( 'wp_sitemaps_max_urls', WP_SITEMAPS_MAX_URLS, $object_type );
 }
+
+if ( ! function_exists( 'esc_xml' ) ) :
+	/**
+	 * Escaping for XML blocks.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @param string $text Text to escape.
+	 * @return string
+	 */
+	function esc_xml( $text ) { // phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals
+		$safe_text = wp_check_invalid_utf8( $text );
+
+		$cdata_regex = '\<\!\[CDATA\[.*?\]\]\>';
+		$regex       = <<<EOF
+/
+	(?=.*?{$cdata_regex})                 # lookahead that will match anything followed by a CDATA Section
+	(?<non_cdata_followed_by_cdata>(.*?)) # the "anything" matched by the lookahead
+	(?<cdata>({$cdata_regex}))            # the CDATA Section matched by the lookahead
+
+|	                                      # alternative
+
+	(?<non_cdata>(.*))                    # non-CDATA Section
+/sx 
+EOF;
+		$safe_text = (string) preg_replace_callback(
+			$regex,
+			function( $matches ) {
+				if ( ! $matches[0] ) {
+					return '';
+				} elseif ( ! empty( $matches['non_cdata'] ) ) {
+					// escape HTML entities in the non-CDATA Section.
+					return esc_xml_non_cdata_section( $matches['non_cdata'] );
+				}
+
+				// Return the CDATA Section unchanged, escape HTML entities in the rest.
+				return esc_xml_non_cdata_section( $matches['non_cdata_followed_by_cdata'] ) . $matches['cdata'];
+			},
+			$safe_text
+		);
+
+		/**
+		 * Filters a string cleaned and escaped for output in XML.
+		 *
+		 * Text passed to esc_xml() is stripped of invalid or special characters
+		 * before output. HTML named character references are converted to their
+		 * equivalent code points.
+		 *
+		 * @since 5.5.0
+		 *
+		 * @param string $safe_text The text after it has been escaped.
+		 * @param string $text      The text prior to being escaped.
+		 */
+		return apply_filters( 'esc_xml', $safe_text, $text ); // phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals
+	}
+endif;
+
+if ( ! function_exists( 'esc_xml_non_cdata_section' ) ) :
+	/**
+	 * Escaping for non-CDATA Section XML blocks.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @param string $text Text to escape.
+	 * @return string
+	 */
+	function esc_xml_non_cdata_section( $text ) { // phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals
```

Comment:
IMO this should be renamed to `_esc_xml_non_cdata_section` and marked with `@access private`,  as its intended for internal use only.

Reply:
sounds good.

Will update shortly.